pkg/matcher/annotation_matcher.go

```diff
@@ -92,6 +92,9 @@ func getAnnotationValues(annotation string) ([]string, error) {
 func getTargetBranch(prun *tektonv1.PipelineRun, event *info.Event) (bool, string, string, error) {
 	var targetEvent, targetBranch string
 	if key, ok := prun.GetObjectMeta().GetAnnotations()[keys.OnEvent]; ok {
+		if key == "[]" {
```

Comment:
@chmouel what about if `on-cel-expression` is defined then these both are not required :thinking: I guess

Reply:
yeah right but then if there is a cel expression the annotation would not even get parsed, here in this case the user does not have cel annotation but specify a empty [] string in the target-branch/event annotation